test/metabase/api/automagic_dashboards_test.clj

```diff
@@ -28,83 +32,107 @@
      ~@body))
 
 (defn- api-call
-  [template & args]
-  (with-rasta
-    (with-dashboard-cleanup
-      (some? ((test-users/user->client :rasta) :get 200 (apply format (str "automagic-dashboards/" template) args))))))
-
-(expect (api-call "table/%s" (data/id :venues)))
-(expect (api-call "table/%s/rule/example/indepth" (data/id :venues)))
+  ([template args] (api-call template args (constantly true)))
+  ([template args revoke-fn]
+   (with-rasta
+     (with-dashboard-cleanup
+       (let [api-endpoint (apply format (str "automagic-dashboards/" template) args)]
+         (and (some? ((test-users/user->client :rasta) :get 200 api-endpoint))
+              (try
+                (do
+                  (perms/delete-related-permissions! (perms-group/all-users) (perms/object-path (data/id)))
+                  (revoke-fn)
+                  (= ((test-users/user->client :rasta) :get 403 api-endpoint)
+                     "You don't have permissions to do that."))
+                (finally
+                  (perms/grant-permissions! (perms-group/all-users) (perms/object-path (data/id)))))))))))
```

Comment:
sorry if I'm being dumb but did you change all the tests that check for 200s and results to revoke perms and check for 403s instead? Instead of adding new tests?

Reply:
I think it would be nice to test *both* that the endpoint works if you have proper permissions, and that it fails if you don't. Instead of just one or the other.

Reply:
otherwise as it is we might end up breaking one of these endpoints and we would never know from the tests because there's nothing to check that they're returning 200s any more

Reply:
It tests for both. Line 40 is the 200 check, below it is the permission check.

Reply:
oh yeah I see now. Cool WFM